onnxruntime/core/providers/cpu/rnn/deep_cpu_gru.cc

```diff
@@ -1024,20 +1034,24 @@ void UniDirectionalGru<T>::Compute(const gsl::span<const T>& inputs_arg,
     }
   }
 
-  if (output_sequence) {
-    // copy last output to final_hidden_state
-    for (int i = 0; i < batch_size_; i++) {
-      const int seq_len = sequence_lengths[i];
+  // copy last output to final_hidden_state
+  for (int i = 0; i < batch_size_; i++) {
+    const int seq_len = sequence_lengths[i];
+    if (seq_len == 0) {
+      auto final_hidden_state_dst = final_hidden_state.begin() + i * hidden_size_;
```

Comment:
We don't need to do the fill if output_sequence is false do we?

Reply:
this one is to fill the Y_h, not Y. I think we need to fill it to keep consistent.